R/mcmc-intervals.R

```diff
@@ -146,274 +218,345 @@ mcmc_areas <- function(x,
-    g_poly <-
-      geom_polygon(data = df_poly, aes_(
-        x = ~ x,
-        y = ~ y,
-        group = ~ name,
-        fill = if (color_by_rhat) ~ rhat else ~ y
-      ))
-
-    # point estimate
-    df_dens$parameter <- df_dens$name
-    pt_data <- dplyr::summarise_(
-      # find y value at which to stop vertical pt est segment
-      dplyr::group_by_(
-        dplyr::left_join(df_dens, data[, c("parameter", "m")],
-                         by = "parameter"),
-        .dots = list(~ parameter)
-      ),
-      .dots = list(maxy = ~ y[which.min(abs(x - m))])
-    )
-    segment_args <- list(
-      data = dplyr::left_join(data, pt_data, by = "parameter"),
-      mapping = aes_(
-        x = ~ m,
-        xend = ~ m,
-        y = ~ y,
-        yend = ~ maxy,
-        color = if (!color_by_rhat) NULL else ~ rhat
-      ),
-      size = 1
-    )
-    if (!color_by_rhat)
-      segment_args$color <- get_color("m")
-    g_point <- do.call("geom_segment", segment_args)
-
-    # bottom line
-    bottom_args <- list(
-      mapping = aes_(
-        x = ~ ll,
-        xend = ~ hh,
-        y = ~ y,
-        yend = ~ y,
-        color = if (!color_by_rhat) NULL else ~ rhat
-      )
-    )
-    if (!color_by_rhat)
-      bottom_args$color <- get_color("d")
-    g_bottom <- do.call("geom_segment", bottom_args)
-
-    graph <- graph + g_poly
-    if (!no_point_est)
-      graph <- graph + g_point
-    graph <- graph + g_bottom + g_dens
-
-    if (color_by_rhat) {
-      graph <- graph +
-        scale_fill_diagnostic("rhat") +
-        scale_color_diagnostic("rhat")
-    } else {
-      graph <- graph + scale_fill_gradient(low = get_color("l"),
-                                           high = get_color("l"),
-                                           guide = "none")
-    }
 
-  } else { # No densities
-
-    # outer interval
-    graph <-
-      graph + geom_segment(aes_(
-        x = ~ ll,
-        xend = ~ hh,
-        y = ~ y,
-        yend = ~ y
-      ),
-      colour = get_color("m")
-    )
-
-    # inner interval
-    segment_args <- list(
-      mapping = aes_(
-        x = ~ l,
-        xend = ~ h,
-        y = ~ y,
-        yend = ~ y,
-        color = if (!color_by_rhat) NULL else ~ rhat
-      ),
-      size = 2,
-      show.legend = FALSE
-    )
-    if (!color_by_rhat)
-      segment_args$color <- get_color("d")
-    graph <- graph + do.call("geom_segment", segment_args)
-
-    # point estimate
-    point_args <- list(
-      mapping = aes_(
-        x = ~ m,
-        y = ~ y,
-        color = if (!color_by_rhat) NULL else ~ rhat,
-        fill = if (!color_by_rhat) NULL else ~ rhat
-      ),
-      size = 4,
-      shape = 21
-    )
-    if (!color_by_rhat) {
-      point_args$color <- get_color("dh")
-      point_args$fill <- get_color("l")
-    }
+# Don't import `filter`: otherwise, you get a warning when using
+# `devtools::load_all(".")` because stats also has a `filter` function
```

Comment:
Yeah that's annoying. It'd be nice to import `filter` from __dplyr__ since we use it a lot but I think it would require importing only what we use from __stats__ as opposed to  just having `@import stats`. We could potentially do that.